### ISSUE TITLE
Feature Request: Next.js App Router support

### ISSUE THREAD

**Issue by n3ps** [coinbase/build-onchain-apps#91]
### Is your feature request related to a problem? Please describe.

For a new project heading into to 2024, at this time I would expect an App Router based Next.js starting point (as opposed to Pages router based)

This is also based on [Next.js's recommendation](https://nextjs.org/docs/app)

### Describe the solution you'd like

Expected: generates an app-based Next.js project

Current: generates a pages-based Next.js project

**Comment by cnasc** [comment]
> at this time I would expect an App Router based Next.js starting point

tl;dr I think if we made App Router the default in this starter kit, we'd end up providing a worse developer experience.

One thing to consider here is that App Router makes some aspects of Web3 development cumbersome. In particular, anything that relies on some client-side state will require a `"use client"` annotation. In a Web3 app almost everything could depend on whether a user's browser extension wallet is connected, is on the right network, has a certain balance, and so on. So at minimum we have some boilerplate burden. On top of that, once you have `"use client"` sprinkled liberally around your codebase, a lot of [the benefits of App Router](https://nextjs.org/docs/app/building-your-application/rendering/server-components#benefits-of-server-rendering) are no longer applicable—we ask users to put up with an admittedly mild itch in exchange for little upside.

Next.js's recommendation to start new apps with App Router is reasonable IMO for most web apps, but not obviously the right choice for our niche. Because [incremental adoption is possible](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration) I think it makes sense to continue with Pages Router for the time being and allow devs to opt-in to server components in the parts of their app where it makes sense.

**Comment by Zizzamia** [comment]
We tested App Router with a few eng who are not Next.js experts, and they got lost very quickly on why `"use client"` was needed. They started question if `wagmi` or `viem` was broken, and eventually, they lost a morning to figure out was just the `"use client"`  story.

I hear your point @n3ps, and I believe in 8 to 12 months, we will switch to App Router when it gets more widely adopted.

However, with previous experience in significant routing changes, there is still time to adopt a new routing system. And on "starting right," I think we have a different point of view. Right, to me, is what has been well received and adopted by the industry.

I know the Next team is working hard on this, and as we see more progress coming up from Next 14, we will decide to upgrade to the latest design pattern.

**Comment by uneeb123** [comment]
Most NextJS tutorials these days recommend App Router. I wasn't even aware that Pages Router has advantages when it comes to web3 apps. Since this is a deliberate decision, it would be cool to read on this in detail somewhere. Since I haven't heard anyone else recommend it yet.

**Comment by Zizzamia** [comment]
I am going to close this one for now, and revisit the Page vs App router decision in six months.

Thank you everyone for participating.
@n3ps @cnasc @uneeb123
